homeassistant/components/litterrobot/switch.py

```diff
@@ -1,68 +1,80 @@
 """Support for Litter-Robot switches."""
+from __future__ import annotations
+
+from typing import Callable
+
 from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
 
 from .const import DOMAIN
-from .hub import LitterRobotEntity
+from .entity import LitterRobotControlEntity
+from .hub import LitterRobotHub
 
 
-class LitterRobotNightLightModeSwitch(LitterRobotEntity, SwitchEntity):
+class LitterRobotNightLightModeSwitch(LitterRobotControlEntity, SwitchEntity):
     """Litter-Robot Night Light Mode Switch."""
 
     @property
-    def is_on(self):
+    def is_on(self) -> bool:
         """Return true if switch is on."""
-        return self.robot.night_light_active
+        return self.robot.night_light_mode_enabled
 
     @property
-    def icon(self):
+    def icon(self) -> str:
         """Return the icon."""
         return "mdi:lightbulb-on" if self.is_on else "mdi:lightbulb-off"
 
-    async def async_turn_on(self, **kwargs):
+    async def async_turn_on(self, **kwargs) -> None:
```

Comment:
```suggestion
    async def async_turn_on(self, **kwargs: Any) -> None:
```

If you add typing, please type the whole line